src/include/rados/librados.hpp

```diff
@@ -95,6 +95,12 @@ namespace librados
     /// move the iterator to a given hash position.  this may (will!) be rounded to the nearest pg.
     uint32_t seek(uint32_t pos);
 
+    /// move the iterator to a given cursor position
+    int seek(const std::string& cursor);
```

Comment:
consider taking `const char*` for the c++ interface too

following this `cursor` argument through the call stack starting from `librados::IoCtx::nobjects_begin(const string& cursor)`, it's passed through `NObjectIterator::seek(const string& cursor)` to `NObjectIteratorImpl::seek(const string& cursor)`, then converted to `const char*` for `rados_nobjects_list_seek_cursor(ctx.get(), cursor.c_str(), nullptr)`, then back to `std::string` for `IoCtxImpl::nlist_seek(const string&)` and `ListCursor::from_str(const string&)`

the substring that `ListCursor::from_str()` passes to `hobject_t::parse()` appears to be the only bit that actually needs to be a `std::string` in this call path

 same applies to `ObjectIterator::seek()` below